main.cc

```diff
@@ -11,6 +11,9 @@
 #include <yaml-cpp/yaml.h>
 
 #include <seastar/util/closeable.hh>
```

Comment:
> main: force user to use specific autorization options in maintenance …
…mode

> In the maintenance mode, it is possible that the node doesn't have system_auth keyspace.
> If so, it won't boot if the user chooses authorization that needs data from this
> keyspace, e.g. roles, passwords.

> Force users to choose authorization options that don't need any data and allow them
> to do all operations.

Maybe maintenance mode should imply this.

If we have a separate listener, then the separate listener could have separate a separate authenticator.

Reply:
+1, using maintenance mode shouldn't require other config changes. Those changes may be forgotten about and the node will re-join the cluster without auth enabled.